scala/scala_android.bzl

```diff
@@ -0,0 +1,66 @@
+load("@io_bazel_rules_scala//scala:scala.bzl", "scala_library")
+
+def scala_android_repositories():
+    native.bind(name = "io_bazel_rules_scala/dependency/android_sdk", actual = "//tools/defaults:android_jar")
+
+def scala_android_library(
+        name,
+        srcs = [],
+        deps = [],
+        resource_files = [],
+        custom_package = None,
+        manifest = None,
+        proguard_specs = [],
+        visibility = None,
+        plugins = [],
+        **kwargs):
+
+    res_deps = []
+    if len(resource_files) > 0:
+        if manifest == None:
+            fail("manifest is required when resource_files are present", "manifest")
+
+        native.android_library(
+            name = name + "_res",
+            custom_package = custom_package,
+            manifest = manifest,
+            resource_files = resource_files,
+            deps = deps,
+            **kwargs
+        )
+        res_deps.append(name + "_res")
+
+    native.android_library(
+        name = name + "_aar",
+        neverlink = 1,
+    )
+
+    scala_library(
+        name = name + "_compile",
+        srcs = srcs,
+        plugins = plugins,
+        deps = deps + ["//external:io_bazel_rules_scala/dependency/android_sdk", name + "_aar"] + res_deps,
+        **kwargs
+    )
+
+    native.java_import(
```

Comment:
why do we need this? does android_library not support the java-sandwich/providers stuff? I thought the point of that was that we could use scala interchangeably with java? Can you add a comment?

Reply:
android_library can't export a scala_library. This is working around that restriction.

What I actually want is to add <name>_compile as one of the exports for the call to android_library. However, if I do that directly I get this error:

    BUILD:462:1: in exports attribute of android_library rule //test:ScalaAndroidLibrary: scala_library rule '//test:ScalaAndroidLibrary_compile' is misplaced here (expected aar_import, android_library, cc_library, java_import, java_library or java_lite_proto_library).

I get the same error using deps (except when I use deps instead of exports I also get an error about how I should use exports instead).

The java_import is a proxy to get around this issue. I'm open to better ways to accomplish this, I admittedly don't have much experience with bazel :)